crates/storage/src/collections/vector.rs

```diff
@@ -0,0 +1,329 @@
+    entries: Entries<V>,
+    /// The storage element for the vector.
+    #[storage]
+    storage: Element,
+}
+
+/// A collection of entries in a vector.
+#[derive(Collection, Clone, Debug, Eq, PartialEq, PartialOrd)]
+#[children(Entry<V>)]
+struct Entries<V> {
+    /// Helper to associate the generic types with the collection.
+    _priv: PhantomData<V>,
+}
+
+/// An entry in a vector.
+#[derive(AtomicUnit, Clone, Debug, Eq, PartialEq, PartialOrd)]
+#[type_id(250)]
+pub struct Entry<V> {
+    /// The index for the entry.
+    index: usize,
+    /// The value for the entry.
+    value: V,
+    /// The storage element for the entry.
+    #[storage]
+    storage: Element,
+}
+
+impl<V: BorshSerialize + BorshDeserialize> Vector<V> {
+    /// Create a new vector collection.
+    ///
+    /// # Errors
+    ///
+    /// If an error occurs when interacting with the storage system, or a child
+    /// [`Element`](crate::entities::Element) cannot be found, an error will be
+    /// returned.
+    ///
+    pub fn new(prefix: &str) -> Result<Self, StoreError> {
+        let mut this = Self {
+            prefix: prefix.to_owned(),
+            entries: Entries::default(),
+            storage: Element::new(
+                &Path::new(format!("::unused::vector::{prefix}::path"))?,
+                None,
+            ),
+        };
+
+        let _ = Interface::save(&mut this)?;
+
+        Ok(this)
+    }
+
+    /// Compute the ID for a element in the vector based on the prefix and index.
+    fn compute_id(&self, index: usize) -> Id {
+        let mut hasher = Sha256::new();
+        hasher.update(self.prefix.as_bytes());
+        hasher.update(index.to_le_bytes());
+        Id::new(hasher.finalize().into())
+    }
+
+    /// Add a value to the end of the vector.
+    ///
+    /// # Errors
+    ///
+    /// If an error occurs when interacting with the storage system, or a child
+    /// [`Element`](crate::entities::Element) cannot be found, an error will be
+    /// returned.
+    ///
+    pub fn push(&mut self, value: V) -> Result<bool, StoreError> {
+        let index = self.len()?;
+        let mut entry = Entry {
+            index,
+            value,
+            storage: Element::new(&self.path(), Some(self.compute_id(index))),
+        };
+
+        Ok(Interface::add_child_to(
+            self.id(),
+            &mut self.entries,
+            &mut entry,
+        )?)
+    }
+
+    /// Remove and return the last value from the vector.
+    ///
+    /// # Errors
+    ///
+    /// If an error occurs when interacting with the storage system, or a child
+    /// [`Element`](crate::entities::Element) cannot be found, an error will be
+    /// returned.
+    ///
+    pub fn pop(&mut self) -> Result<Option<V>, StoreError> {
+        let len = self.len()?;
+        if len == 0 {
+            return Ok(None);
+        }
+
+        let index = len - 1;
+        let id = self.compute_id(index);
+        let value = self.get(index)?;
```

Comment:
`child_info_for` returns a list of `Id`s, we can get the ID of that last item from there